packages/package-json-lint/src/rules/requireVersionSet.js

```diff
@@ -5,7 +5,7 @@ module.exports = ({
 }) => {
   const currentProblems = versionSet.map(({ name, versionRange }) => {
     const dependencyVersion = dependencies[name];
-    if (dependencyVersion && !semver.intersects(dependencyVersion, versionRange)) {
+    if (dependencyVersion && !semver.intersects(dependencyVersion, versionRange) && !(ruleConfig.severity.allowList && ruleConfig.severity.allowList.includes(name))) {
```

Comment:
as we add more rules will this if continue to grow? Is there a way we could clean this up?